src/pwr.ps1

```diff
@@ -87,81 +93,101 @@ Class SemanticVersion : System.IComparable {
-		if ($tag -match $pattern) {
-			$this.Major = if ($Matches[1]) { $Matches[1] } else { 0 }
-			$this.Minor = if ($Matches[2]) { $Matches[2] } else { 0 }
-			$this.Patch = if ($Matches[3]) { $Matches[3] } else { 0 }
-			$this.Build = if ($Matches[4]) { $Matches[4].Substring(1) } else { 0 }
+	hidden init([string]$Tag, [string]$Pattern) {
+		if ($Tag -match $Pattern) {
+			$This.Major = if ($Matches[1]) { $Matches[1] } else { 0 }
+			$This.Minor = if ($Matches[2]) { $Matches[2] } else { 0 }
+			$This.Patch = if ($Matches[3]) { $Matches[3] } else { 0 }
+			$This.Build = if ($Matches[4]) { $Matches[4].Substring(1) } else { 0 }
 		}
 	}
 
-	SemanticVersion([string]$tag, [string]$pattern) {
-		$this.init($tag, $pattern)
+	SemanticVersion([string]$Tag, [string]$Pattern) {
+		$This.init($Tag, $Pattern)
 	}
 
-	SemanticVersion([string]$version) {
-		$this.init($version, '^([0-9]+)\.([0-9]+)\.([0-9]+)(\+[0-9]+)?$')
+	SemanticVersion([string]$Version) {
+		$This.init($Version, '^([0-9]+)\.([0-9]+)\.([0-9]+)(\+[0-9]+)?$')
 	}
 
 	SemanticVersion() { }
 
 	[bool] LaterThan([object]$Obj) {
-		return $this.CompareTo($obj) -gt 0
+		return $This.CompareTo($Obj) -gt 0
 	}
 
 	[int] CompareTo([object]$Obj) {
-		if ($Obj -isnot $this.GetType()) {
-			throw "cannot compare types $($Obj.GetType()) and $($this.GetType())"
-		} elseif ($this.Major -ne $Obj.Major) {
-			return $this.Major - $Obj.Major
-		} elseif ($this.Minor -ne $Obj.Minor) {
-			return $this.Minor - $Obj.Minor
-		} elseif ($this.Patch -ne $Obj.Patch) {
-			return $this.Patch - $Obj.Patch
+		if ($Obj -isnot $This.GetType()) {
+			throw "cannot compare types $($Obj.GetType()) and $($This.GetType())"
+		} elseif ($This.Major -ne $Obj.Major) {
+			return $This.Major - $Obj.Major
+		} elseif ($This.Minor -ne $Obj.Minor) {
+			return $This.Minor - $Obj.Minor
+		} elseif ($This.Patch -ne $Obj.Patch) {
+			return $This.Patch - $Obj.Patch
 		} else {
-			return $this.Build - $Obj.Build
+			return $This.Build - $Obj.Build
 		}
 	}
 
 	[string] ToString() {
-		return "$($this.Major).$($this.Minor).$($this.Patch)$(if ($this.Build) {"+$($this.Build)"})"
+		return "$($This.Major).$($This.Minor).$($This.Patch)$(if ($This.Build) {"+$($This.Build)"})"
 	}
 
 }
 
 function global:Prompt {
 	if ($env:InPwrShell) {
-		Write-Host -ForegroundColor Blue -NoNewline 'pwr:'
-		Write-Host " $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1))" -NoNewline
+		Write-PwrHost -ForegroundColor Blue -NoNewline 'pwr:'
+		Write-PwrHost " $($ExecutionContext.SessionState.Path.CurrentLocation)$('>' * ($NestedPromptLevel + 1))" -NoNewline
 		return ' '
 	} else {
-		"PS $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1)) "
+		"PS $($ExecutionContext.SessionState.Path.CurrentLocation)$('>' * ($NestedPromptLevel + 1)) "
 	}
 }
 
-function Invoke-PwrWebRequest($Uri, $Headers, $OutFile, [switch]$UseBasicParsing) {
-	if ($Offline) {
```

Comment:
This was here for good reason. Without this you need to always know where web requests happen and have offline blocks for it. Case in point: `pwr update` now succeeds with `-offline` specified.

Reply:
Then we need to do that. As it was, this function would just use `-Offline` to spit out this error message, then continue executing the web request. This is not what we want, when you specify `-Offline` it should never try to hit the network.

Reply:
I'll test this on main, but it shouldn't have. `ErrorActionProgress` is `Stop` meaning the write-error alone should have stopped execution.

Reply:
It doesn't return after this error message, and `$Offline` wasn't gating anything from accessing this function previously.

Reply:
It did not continue on `main`. I'd be curious to know how you produced that behavior.

Reply:
Granted, the console was ugly, but the desired behavior was there. Agree that the feedback to the user needed improvement.

Reply:
Maybe I'm missing, does `Write-Error` throw an exception?

Reply:
Looks like it's equivalent to `throw` with more features.

Reply:
Yeah, it's basically a throw, but I set the pwr script to immediately stop execution when it sees an error. Our new write-fatal is close to equivalent